pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -139,7 +158,26 @@ func init() {
 // NewPodIpMetricsCollector returns a new Collector exposing pod IP allocation
 // stats.
 func NewPodIPMetricsCollector() (Collector, error) {
-	return &podIPMetricsCollector{clock: &realClock{}}, nil
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error creating in-cluster config: %v", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("error creating clientset: %v", err)
+	}
+
+	nodeName := os.Getenv("HOSTNAME")
```

Comment:
`os.Hostname()` or `k8s.io/component-helpers/node/util.GetHostname("")` (this runs some canonicalization)

Reply:
Done.